src/js/controllers/ctrl-user-notification.js

```diff
@@ -26,7 +26,44 @@ app.controller('UserNotificationCtrl', function (Please, $scope, $sce) {
 	$scope.toggleOpen = _toggleOpen
 
 	Materia.Coms.Json.send('notifications_get').then((notifications) => {
-		$scope.notifications = notifications
+		if (!notifications && !notifications.length) return
+		$scope.notifications = notifications.map((notification) => {
+			// does a notification include an action? Perform some additional setup based on the action performed. This is intended to be extensible.
+			if (notification.action) {
+				// someone is requesting access to the widget, enable the "add a collaborator" button and callback behavior
+				switch (notification.action) {
+					case 'access_request':
+						notification.button_action_text = 'Add as Collaborator'
+						notification.button_action_callback = () => {
+							var url = new URL(window.location.href)
+
+							// already on the my-widgets page; don't need to redirect, just send a broadcast with relevant info
+							if (
+								url.pathname.match(/^(\/my-widgets)/) &&
+								url.pathname.match(/^(\/my-widgets)/).length
```

Comment:
Might store `url.pathname.match(/^(\/my-widgets)/)` in a variable one line up so that it only has to run the regex once, but that's a minor nit.